client/components/post-schedule/clock.jsx

```diff
@@ -48,6 +48,11 @@ class PostScheduleClock extends Component {
 
 		let value = this.maybeConvertHour( Number( event.target.value ) ) - operation;
 
+		// Snap back a day, when looping through date line.
+		if ( 0 === value && 1 === operation ) {
+			modifiers.date = this.props.date.get( 'date' ) - 1;
```

Comment:
I'm not sure if is ok add `date` key into the modifiers. I think we should just keep working with `hour` and `minute`. Why not just reset the hour value?
Beyond of this, I've found this bug:

`Initial date`: Day 15
`hour`: decrements from `1` to `0`

![image](https://cloud.githubusercontent.com/assets/77539/22928433/2f5b1522-f295-11e6-926e-bc8495670b5e.png)

Day number shouldn't change?